block/bpf_io_filter.c

```diff
@@ -0,0 +1,229 @@
+#include <linux/bpf_io_filter.h>
+
+#include <linux/bpf.h>
+#include <linux/bpf_trace.h>
+#include <linux/filter.h>
+#include <linux/btf.h>
+#include <linux/filter.h>
+#include <linux/kallsyms.h>
+#include <linux/bpf_verifier.h>
+#include <linux/kobject.h>
+#include <linux/sysfs.h>
+#include <linux/genhd.h>
+#include <uapi/linux/bpf.h>
+
+#include "blk-bpf-io-filter.h"
+
+static struct kobject *bpf_io_filter_kobj;
+static struct kset *progs_kset;
+
+static struct attribute *event_attrs[] = {
+	NULL,
+};
+const struct bpf_func_proto *bpf_tracing_func_proto(
+	enum bpf_func_id func_id, const struct bpf_prog *prog);
+
+static const struct attribute_group event_attr_group = {
+	.attrs = event_attrs,
+	.name = "events",
+};
+
+static int __init bpf_io_filter_init(void)
+{
+	int err = -ENOMEM;
+
+	pr_info("Initializing bpf_io_filter");
+
+	bpf_io_filter_kobj = kobject_create_and_add("bpf_io_filter",
+						    kernel_kobj);
+	if (!bpf_io_filter_kobj)
+		goto exit;
+
+	err = sysfs_create_group(bpf_io_filter_kobj, &event_attr_group);
+	if (err)
+		goto events_exit;
+
+	progs_kset = kset_create_and_add("progs", NULL, bpf_io_filter_kobj);
```

Comment:
How are we planning to use this?

Reply:
I am not quite sure, I did not have to modify this at all. Any ideas @srnym ?

Reply:
I dont recall the need for this.

Reply:
Removed init function and everything is still working fine, can add back in later if needed